app/main.py

```diff
@@ -2,5 +2,20 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def wrapper(*args, **kwargs) -> Callable:
+        if any(isinstance(arg, (list, dict, set)) for arg in args):
+            raise ValueError("All arguments must be immutable")
```

Comment:
The check for immutability only covers the positional arguments (`args`). You should also check the keyword arguments (`kwargs`) to ensure they are immutable. Consider iterating over `kwargs.values()` as well.